kuma/javascript/ssr-server.js

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable no-console */
+/*
+ * This is a simple Express server for performing server side rendering
+ * of our React UI. The main endpoint is POST /ssr, and it invokes the
+ * ssr() function from dist/ssr.js (which is webpacked from src/ssr.jsx).
+ */
+const express = require('express'); // Express server framework
+const morgan = require('morgan');
+
+const ssr = require('./dist/ssr.js'); // Function to do server-side rendering
+
+// Configuration
+const PID = process.pid;
+const PORT = parseInt(process.env.SSR_PORT) || 8000;
+const MAX_BODY_SIZE = 1024 * 1024; // 1 megabyte max JSON payload
+
+// Start New Relic logging if it is configured
+if (process.env.NEW_RELIC_LICENSE_KEY && process.env.NEW_RELIC_APP_NAME) {
+    console.log('Starting New Relic logging for KumaScript.');
+    require('newrelic');
+}
+
+// We're using the Express server framework
+const app = express();
+
+// Log all requests, so we get timing data for SSR.
+app.use(morgan('tiny'));
+
+// Handle JSON payloads in POST requests, putting data in req.body
+app.use(express.json({ limit: MAX_BODY_SIZE }));
+
+// A hello world endpoint for easy verification that the service is running
+app.get('/', (req, res) => {
+    res.send('<html><body><p>SSR server ready</p></body></html>');
+});
+
+// revision, health and readiness endpoints used (I think) by Kubernetes
+app.get('/revision/?', (req, res) => {
```

Comment:
I don't think this endpoint is needed, but it doesn't hurt, so let's keep it, at least for now.

Reply:
I copied it from KumaScript. Seemed like the idea of being able to query what version is running might be useful to someone at some point.